sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -989,7 +989,7 @@ class Analyzer(
               withPosition(u) {
                 try {
                   outer.resolve(nameParts, resolver) match {
-                    case Some(outerAttr) => OuterReference(outerAttr)
+                    case Some(outerAttr) => OuterReference(outerAttr)()
```

Comment:
The `LEFT ANTI` join should produce the correct result. Unfortunately we push down the `tbl_a.c1 = tbl_a.c2` expression into the `tbl_a` side of the plan. So we need to fix this. I have created https://issues.apache.org/jira/browse/SPARK-18597 to track this.